marshaler_example_test.go

```diff
@@ -62,7 +62,7 @@ func ExampleMarshaler() {
 		},
 	}
 
-	logger := zap.New(zap.NewJSONEncoder(zap.NoTime()))
+	logger := zap.New(zap.WriterFacility(zap.NewJSONEncoder(zap.NoTime()), nil, zap.InfoLevel))
```

Comment:
Agreed; this PR is against dev, and takes us to a temporary "uncanny valley" on API ergonomics. Either this PR will grow in scope to move all the chairs and rebuild the public API layer, or we'll do that in a Soon ™ to come additional dev-based PR.